profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackFramesCollectorBase.cpp

```diff
@@ -148,6 +152,8 @@ void StackFramesCollectorBase::PrepareForNextCollection()
     // This is because malloc() uses a lock and so if we suspend a thread that was allocating, we will deadlock.
     // So we pre-allocate the memory buffer and reset it before suspending the target thread.
     _pStackSnapshotResult->Reset();
+    _pStackSnapshotResult->SetCallstack(_callstackPool->Get());
```

Comment:
In case of memory limited CallStack (like ringbuffer-based), we might want to stop here so maybe the method should return a bool. Otherwise, we will have to wait for the stackwalking code to check later on

Reply:
👍